client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -0,0 +1,124 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import java.security.{MessageDigest, DigestInputStream}
+import java.io.{File, FileInputStream}
+
+import scala.collection.mutable.HashMap
+import scala.util.{Failure, Success, Try}
+
+/** Captures metadata about files
+ *  @param paths list of filepaths or directory paths
+ */
+ case class PathBlob(private val paths: List[String]) extends Dataset {
+  private val BufferSize = 8192
+
+  private val metadataList = paths.map(expanduser)
+  .flatMap((path: String) => processPath(new File(path)))
+  .map(metadata => metadata.path -> metadata)
```

Comment:
Can we have a more scala-like indentation? This is pretty hard to read.

Reply:
Yeah atom's autoindent betrayed me there. I've added one indentation per chain command. Hopefully that helps

Reply:
Wait I found multiple method chaining styles via googling. I think this will look best:
https://alvinalexander.com/scala/how-to-fluent-style-programming-methods-in-scala/